### PR TITLE
Git Data Model Section

### PR DESCRIPTION
**Pull request by martyschaer**
- Add section about internal data model.
- Please feel free to point out any areas that are unclear, so they can be improved.

### CODE REVIEW

**Comment by martyschaer** [comment]
Improvement: describe how commits refer to their parents, and how branches and tags work.

File: handout.md [hunk 1]
```diff
@@ -169,7 +175,127 @@ the main branch.
...
+
+New file_a.txt created
+```
+
+Above you can see the contents of a commit.
+From bottom to top the content is as follows:
+The commit message you entered,
+who created the commit,
+who authored the commit
+^[Author and Committer can be different when merges etc. get involved].
```
Comment (cyrillbolliger) [inline handout.md]: The dot should probably be inside the brackets...?

File: handout.md [hunk 2]
```diff
@@ -169,7 +175,127 @@ the main branch.
...
+```{.sh}
+# Some of the IDs are shortened, indicated by '...'
+$ git cat-file -t 94d90800...
+tree
+$ g cat-file -p 94d90800..
+100644 blob 6434b133...    file_a.txt
+```
+
+So in order, the tree object contains the following:
+The file mode ^[indicating the file permissions],
```
Comment (cyrillbolliger) [inline handout.md]: "The" should be lower case: https://www.grammarly.com/blog/capitalization-after-colons/

File: handout.md [hunk 3]
```diff
@@ -169,7 +175,127 @@ the main branch.
...
+of 'Pro Git'] commands.
+
+### Objects
+Git, at it's core, is a _content addressable file system_;
+meaning you can give it any kind of content and it will return a key
+which you can later use to retrieve that content.
+Git stores it's content in _objects_.
+
+__`blob`__  
+When you `git add` a file a _blob_ object is created and it gets added to the index.
```
Comment (cyrillbolliger) [inline handout.md:284]: Should we introduce the _index_ somewhere?

**Review by cyrillbolliger — CHANGES_REQUESTED** [review]
- Excellent intro, thanks a lot for writing!
- As the handout should probably meet some scientific criteria, we should not use opinionated words

File: handout.md [hunk 4]
```diff
@@ -3,9 +3,15 @@ title: Maven & Git
 titlepage: false
 lot: false
 toc: false
+author:
+- Bolliger Cyrill Alexander
```
Comment (cyrillbolliger) [inline handout.md]:
```suggestion
- Bolliger Cyrill
```

File: handout.md [hunk 5]
```diff
@@ -169,7 +175,127 @@ the main branch.
...
+100644 blob 47c6340d...      simplegit.rb
+```
+
+Which represents a structure as seen in this image:
+
+![](img/data-model.png)
+
+^[Image Credit: [git-scm.com](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects)]
+
+Cleverly, git only needs to update the (sub-)trees that actually change with
```
Comment (cyrillbolliger) [inline handout.md]: Maybe remove _Cleverly_ to be more scientific?

File: handout.md [hunk 6]
```diff
@@ -169,7 +175,127 @@ the main branch.
 
 ## Collaboration
 
+\newpage
+
 ## Data Model
+Git's internal data model is extremely simple and elegant,
```
Comment (cyrillbolliger) [inline handout.md]: I thinks it's less opinionated if we write "Git's internal data model is _designed to be_ extremely simple and elegant"

**Comment by martyschaer** [comment]
>     * Excellent intro, thanks a lot for writing!
> 
>     * As the handout should probably meet some scientific criteria, we should not use opinionated words

I think that's a great idea. I will look into resolving the issues in this either tomorrow or on Wednesday.
(We still have until the 2020-10-22 until the handout needs to be finished).

**Review by cyrillbolliger — APPROVED** [review]
Thanks a lot!

**Comment by martyschaer** [comment]
@cyrillbolliger I've added a section about branches in the data model section.
With that, I think this branch is ready to merge.

**Comment by martyschaer** [comment]
Build Job fails right now when trying to upload the PDF, possibly related to actions/upload-artifact/issues/136
Edit: Worked for now.

**Pull request merged**